src/pip/_internal/models/link.py

```diff
@@ -248,6 +248,11 @@ def from_json(
         yanked_reason = file_data.get("yanked")
         dist_info_metadata = file_data.get("dist-info-metadata")
         hashes = file_data.get("hashes", {})
+        link_hash = None
+        for hash_name, hash_value in hashes.items():
+            if hash_name in _SUPPORTED_HASHES:
+                link_hash = LinkHash(name=hash_name, value=hash_value)
+                break
```

Comment:
Perhaps we should get the first hash in _SUPPORTED_HASHES so as to get a deterministic result that does not depend on the order of hashes provided by the index?

Reply:
That makes sense. If the index starts to support newer hashes we'll end up orphaning cache entries, but we'll only do so when the hashes become stronger (due to the order of _SUPPORTED_HASHES entries ) and not randomly due to index changes.

For updates like this do you prefer a follow on commit or a squash update and force push?

Reply:
I tend to find squashes cleaner (you can click force-pushed to see what changed), but I don't think we have any strong enforced preferences in the code base.